### ISSUE TITLE
STAKE instruction and mechanism in smart contract

### ISSUE THREAD

**Issue by taihaofu** [tronprotocol/tips#171]
## Background

As mentioned in [156](https://github.com/tronprotocol/tips/issues/156) and [157](https://github.com/tronprotocol/tips/issues/157), TRON need to provide vote and collect SR reward mechanism for smart contract. Motivate contract developer and increase the profit for TRON ecosystem contributors. But FREEZE/VOTE system is not direct for a smart contract. This issue provide a more efficient and direct way to implement a similar functionality for smart contracts.

## Implementation

Basically we need at least 2 Instructions:
* `STAKE`           (update stake and vote for a single SR)
* `UNSTAKE`     (cancel stake)

`STAKE` only available for single SR, and update the whole vote & account state on chain, which means smart contract will only allow single SR staking change in one transaction. 

assembly level:
`stake(sr_address, trx_amount)`
`unstake()`

**Comment by yanghang8612** [comment]
The related [TIP-171](https://github.com/tronprotocol/tips/blob/master/tip-171.md) has been marked as `Withdrawn` which means that we do not have further implemention plan.
So we close this issue.
